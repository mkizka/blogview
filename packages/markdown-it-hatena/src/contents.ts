import * as MarkdownIt from "markdown-it";

export const contentsPlugin: MarkdownIt.PluginSimple = (md) => {
  md.use(require("markdown-it-anchor"));
  md.use(require("markdown-it-table-of-contents"), {
    markerPattern: /^\[:contents\]/i,
    includeLevel: [1, 2, 3, 4, 5, 6],
  });
};
