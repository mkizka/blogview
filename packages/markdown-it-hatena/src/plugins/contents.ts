import type { PluginSimple } from "markdown-it";

export const contentsPlugin: PluginSimple = (md) => {
  md.use(require("markdown-it-anchor"));
  md.use(require("markdown-it-table-of-contents"), {
    markerPattern: /^\[:contents\]/i,
    includeLevel: [1, 2, 3, 4, 5, 6],
  });
};
