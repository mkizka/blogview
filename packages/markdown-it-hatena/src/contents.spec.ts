import * as MarkdownIt from "markdown-it";
import { contentsPlugin } from "./contents";

const md = new MarkdownIt().use(contentsPlugin);

const contents = `[:contents]
# 見出し1
## 見出し2
### 見出し3
`;

test("目次記法[:contents]の対応", () => {
  expect(md.render(contents)).toMatchSnapshot();
});
