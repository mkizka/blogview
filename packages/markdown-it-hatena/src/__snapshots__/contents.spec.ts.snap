// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`目次記法[:contents]の対応 1`] = `
"<p><div class=\\"table-of-contents\\"><ul><li><a href=\\"#%E8%A6%8B%E5%87%BA%E3%81%971\\">見出し1</a><ul><li><a href=\\"#%E8%A6%8B%E5%87%BA%E3%81%972\\">見出し2</a></li></ul></li></ul></div></p>
<h1 id=\\"%E8%A6%8B%E5%87%BA%E3%81%971\\" tabindex=\\"-1\\">見出し1</h1>
<h2 id=\\"%E8%A6%8B%E5%87%BA%E3%81%972\\" tabindex=\\"-1\\">見出し2</h2>
<h3 id=\\"%E8%A6%8B%E5%87%BA%E3%81%973\\" tabindex=\\"-1\\">見出し3</h3>
"
`;
